app/controllers/change_requests_controller.rb

```diff
@@ -53,6 +55,79 @@ def create_change_request
     render status: :created, json: ChangeRequestsPresenter.present(@change_request)
   end
 
+  def handle_insert_change_request
+    resource_id = params[:parent_resource_id]
+    change_request = params[:change_request]
+    case params[:type]
+    when "addresses"
+      puts "AddressInsertChangeRequest"
+      address = Address.new
+      permitted_fields = [
+        :attention,
+        :address_1,
+        :address_2,
+        :address_3,
+        :address_4,
+        :city,
+        :state_province,
+        :postal_code,
+        :country,
+        :online,
+        :region,
+        :name,
+        :description,
+        :transportation
+      ]
+      required_fields = [
+        :address_1, 
+        :city, 
+        :state_province, 
+        :postal_code, 
+        :country
+      ]
+      change_request_params = change_request.require(:field_changes).permit(permitted_fields)
+      change_request_params.require(required_fields)
+      address.attributes = change_request_params
+      address.resource_id = resource_id
+
+      # Performing geoencoding here because we want to save the address before we hit persist_change
+      begin
+        a = Geokit::Geocoders::GoogleGeocoder.geocode address.address_1 + "," + address.city + "," + address.state_province
+        address.latitude = a.latitude
+        address.longitude = a.longitude
+
+      rescue => error
+        puts "google geocoding failed for address " + address.id.to_s + ": " + error.message
+
+      end
```

Comment:
Since this block of code is mostly the same as the other instance of it in this file, could you refactor it into a separate method that both locations call? I think the one part that is different is that in this block, you directly modify the `Address` object, but in the other block, it directly modifies the `field_changes` hash, so instead, I think you can just return the output of `GoogleGeocoder.geocode`.